packages/basic/index.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+import { existsSync } from 'node:fs'
+import { readFile } from 'node:fs/promises'
+import { resolve } from 'node:path'
+
+import { ConfigManager } from '@platformatic/config'
+
+import { packageJson, schema } from './lib/schema.js'
+import { ServerStackable } from './lib/server.js'
+
+const validFields = [
+  'main',
+  'exports',
+  'exports',
+  'exports#node',
+  'exports#import',
+  'exports#require',
+  'exports#default',
+  'exports#.#node',
+  'exports#.#import',
+  'exports#.#require',
+  'exports#.#default',
+]
+
+const validFilesBasenames = ['index', 'main', 'app', 'application', 'server', 'start', 'bundle', 'run', 'entrypoint']
+
+async function parsePackageJson (root) {
+  let entrypoint
+  let packageJson
+  let hadEntrypointField = false
+
+  try {
+    packageJson = JSON.parse(await readFile(resolve(root, 'package.json'), 'utf-8'))
+  } catch {
+    // No package.json, we only load the index.js file
+    packageJson = {}
+  }
+
+  for (const field of validFields) {
+    let current = packageJson
+    const sequence = field.split('#')
+
+    while (current && sequence.length && typeof current !== 'string') {
+      current = current[sequence.shift()]
+    }
+
+    if (typeof current === 'string') {
+      entrypoint = current
+      hadEntrypointField = true
+      break
+    }
+  }
+
+  if (!entrypoint) {
+    for (const basename of validFilesBasenames) {
+      for (const ext of ['js', 'mjs', 'cjs']) {
+        const file = `${basename}.${ext}`
+
+        if (existsSync(resolve(root, file))) {
+          entrypoint = file
+          break
+        }
+      }
+
+      if (entrypoint) {
+        break
+      }
+    }
+  }
+
+  return { packageJson, entrypoint, hadEntrypointField }
+}
+
+function transformConfig () {
+  if (this.current.watch === undefined) {
+    this.current.watch = { enabled: false }
+  }
+
+  if (typeof this.current.watch !== 'object') {
+    this.current.watch = { enabled: this.current.watch || false }
+  }
+}
+
+export async function buildStackable (opts) {
+  const root = opts.context.directory
```

Comment:
```suggestion
  const root = dirname(opts.config)
```

Reply:
It wouldn't work: in some situation (for instance a bare Node app) you have no config file at all.

Reply:
Ohh, I see. We need to change the way we load a stackable than.

Reply:
I already did in this PR (https://github.com/platformatic/platformatic/pull/2932/files#diff-7e30935e0bd9b1f4e126148d7cef2e54f77ba2ca0575602256c45f6cb561c6b1). Is there anything else you think it's missing?

Reply:
No, looks good to me. Thanks.